TrafficCapture/coreUtilities/src/main/java/org/opensearch/migrations/tracing/RootOtelContext.java

```diff
@@ -0,0 +1,139 @@
+                        .setTimeout(2, TimeUnit.SECONDS)
+                        .build())
+                .setScheduleDelay(100, TimeUnit.MILLISECONDS)
+                .build();
+        final var metricReader = PeriodicMetricReader.builder(OtlpGrpcMetricExporter.builder()
+                        .setEndpoint(collectorEndpoint)
+                        // see https://opentelemetry.io/docs/specs/otel/metrics/sdk_exporters/prometheus/
+                        // "A Prometheus Exporter MUST only support Cumulative Temporality."
+                        //.setAggregationTemporalitySelector(AggregationTemporalitySelector.deltaPreferred())
+                        .build())
+                .setInterval(Duration.ofMillis(1000))
+                .build();
+        final var logProcessor = BatchLogRecordProcessor.builder(OtlpGrpcLogRecordExporter.builder()
+                        .setEndpoint(collectorEndpoint)
+                        .build())
+                .build();
+
+        var openTelemetrySdk = OpenTelemetrySdk.builder()
+                .setTracerProvider(SdkTracerProvider.builder().setResource(serviceResource)
+                        .addSpanProcessor(spanProcessor).build())
+                .setMeterProvider(SdkMeterProvider.builder().setResource(serviceResource)
+                        .registerMetricReader(metricReader).build())
+                .setLoggerProvider(SdkLoggerProvider.builder().setResource(serviceResource)
+                        .addLogRecordProcessor(logProcessor).build())
+                .build();
+
+        // Add hook to close SDK, which flushes logs
+        Runtime.getRuntime().addShutdownHook(new Thread(openTelemetrySdk::close));
+        return openTelemetrySdk;
+    }
+
+    public static OpenTelemetry initializeOpenTelemetry(String collectorEndpoint, String serviceName) {
+        return Optional.ofNullable(collectorEndpoint)
+                .map(endpoint -> initializeOpenTelemetryForCollector(endpoint, serviceName))
+                .orElse(OpenTelemetrySdk.builder().build());
+    }
+
+
+    public RootOtelContext(String scopeName) {
+        this(scopeName, null);
+    }
+
+    public RootOtelContext(String scopeName, String collectorEndpoint, String serviceName) {
+        this(scopeName, initializeOpenTelemetry(collectorEndpoint, serviceName));
+    }
+
+    public RootOtelContext(String scopeName, OpenTelemetry sdk) {
+        openTelemetryImpl = sdk != null ? sdk : initializeOpenTelemetry(null, null);
+        this.scopeName = scopeName;
+    }
+
+    @Override
+    public RootOtelContext getEnclosingScope() {
+        return null;
+    }
+
+    OpenTelemetry getOpenTelemetry() {
+        return openTelemetryImpl;
+    }
+
+    @Override
+    public MeterProvider getMeterProvider() {
+        return getOpenTelemetry().getMeterProvider();
+    }
+
+    @Override
+    public AttributesBuilder fillAttributes(AttributesBuilder builder) {
+        return builder; // nothing more to do
```

Comment:
For consistency, should we call the fillAttributes method from IRootOtelContext/IInstrumentationAttributes? 

Also, do we need to set the hadException attribute?

Reply:
Hmmmm... we could call fillAttributes, but this class is a very special case of an attributes object.  It serves as the base case so that it can be the root object of a hierarchy of contexts to give everybody in the tree access to its functionality.  However, if the interface's fillAttributes function is modified (even slightly), it will be to support a general case, not a one-off case.  
My take is that this is a tie.  Do you think that there's an argument that we should be calling the interface's default impl?